File: backend/ModuMessenger/src/main/java/com/example/modumessenger/chat/entity/ChatRoom.java
package com.example.modumessenger.chat.entity;

import com.example.modumessenger.common.domain.BaseTimeEntity;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ChatRoom extends BaseTimeEntity {

    @Id
    @Column(name = "chat_room_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String roomId;

    @Column(nullable = false)
    private String roomName;

    @Column(nullable = false)
    private String roomImage;

    @Column(nullable = false)
    private String lastChatMsg;

    @Column(nullable = false)
    private String lastChatTime;

    @ElementCollection
    @CollectionTable(name = "members", joinColumns = @JoinColumn(name = "member_id"))
    @Column(name = "members")
    private List<String> userIds;

    @OneToMany(mappedBy = "chatRoom", cascade = CascadeType.ALL)
    @ToString.Exclude
    private List<Chat> chat;

    public ChatRoom(String roomName) { this.roomName = roomName; }

    public ChatRoom(String roomId, String roomName, String roomImage, String lastChatMsg) {
        this.roomId = roomId;
        this.roomName = roomName;
        this.roomImage = roomImage;
        this.lastChatMsg = lastChatMsg;
        this.lastChatTime = LocalDateTime.now().toString();
    }
}

File: android/modu-chat/app/src/main/kotlin/com/example/modumessenger/core/model/ChatRoom.kt
package com.example.modumessenger.core.model

/** 채팅방. [unreadCount] 는 서버 방 응답에 없고 unread API 와 소켓 수신으로 클라이언트가 채운다. */
data class ChatRoom(
    val roomId: String = "",
    val roomName: String = "",
    val roomImage: String = "",
    val lastChatMsg: String = "",
    val lastChatId: String = "",
    val lastChatTime: String = "",
    val members: List<Member> = emptyList(),
    val unreadCount: Int = 0,
)
